### ISSUE TITLE
no-irregular-whitespace: irregular whitespace inside comments is never reported (skipComments: false default not implemented)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25616]
## Summary

oxlint's `no-irregular-whitespace` rule does not scan comment tokens, so irregular whitespace (e.g. U+200B ZERO WIDTH SPACE) inside a `//` or `/* */` comment is never reported. ESLint flags this by default because the rule's `skipComments` option defaults to `false` (only `skipStrings` defaults to `true`). As a side effect, an `eslint-disable-next-line no-irregular-whitespace` directive suppressing such a comment is reported as unused under `--report-unused-disable-directives`.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.ts` — the invisible character between the quotes in the trailing comment is a literal U+200B ZERO WIDTH SPACE (bytes `e2 80 8b`):

```ts
export const ZERO_SPACE = String.fromCharCode(0x200b) // '​'
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"no-irregular-whitespace": ["error"]}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config --report-unused-disable-directives repro.ts
```

## Expected

The irregular whitespace inside the trailing comment is flagged, because `no-irregular-whitespace` defaults to `skipComments: false`. ESLint 7 and ESLint 9 both report:

```
1:59  error  Irregular whitespace not allowed  no-irregular-whitespace
```

## Actual

oxlint 1.78.0 reports nothing for the comment. Prefixing the line with `// eslint-disable-next-line no-irregular-whitespace` makes it report instead:

```
repro.ts:1:1: warning: Unused eslint-disable directive (no problems were reported).
```

i.e. the rule never inspected the comment token, so it found no violation and considered the suppression unused.

The rule logic itself works on non-comment code: U+00A0 in a code position (`export const CONTROL = 42` with a U+00A0 before `42`) is correctly flagged at `1:23` by oxlint and both ESLint versions, and U+200B inside a string literal is correctly skipped everywhere via the default `skipStrings: true`. The divergence is specifically the comment scanning path.

## Cross-checked

| Linter | U+200B in comment | disable directive | U+00A0 in code (control) | U+200B in string |
| --- | --- | --- | --- | --- |
| ESLint 7.23.0 (`@typescript-eslint/parser` 6.21.0) | error at 1:59 | treated as used (silent) | error at 1:23 | skipped |
| ESLint 9.39.4 flat config (typescript-eslint 8.62.1) | error at 1:59 | treated as used (silent) | error at 1:23 | skipped |
| oxlint 1.78.0 | not reported | reported as unused | error at 1:23 | skipped |
